pelican_bib/bib.py

```diff
@@ -160,6 +201,159 @@ def add_publications(generator):
     generator.context['publications_lists'] = publications_lists
 
 
+class Bibliography(Directive):
+    """
+    Directive to embed bibliographies into articles/posts.
+
+    Usage::
+
+        .. bibliography:: PUBLICATIONS_SRC
+          :template: TEMPLATE_NAME (optional)
+          :options: TEMPLATE_OPTIONS_DICT (optional)
+          :class: CLASS_NAME (optional)
+          :filter_tag: FILTER_TAG (optional)
+          :pybtex_style_args: PYBTEX_STYLE_ARGS (optional)
+          :sorting_style: PYBTEX_SORTING_STYLE (optional)
+          :abbreviate_names: PYBTEX_ABBREVIATE_NAMES (optional)
+          :name_style: PYBTEX_NAME_STYLE (optional)
+
+    e.g.::
+
+        .. bibliography:: osm.bib
+          :template: publications_unsrt
+          :options: {'groupby_value': year}
+          :class: my-bib
+          :pybtex_style_args: {'sorting_style': 'author_year_title',
+                               'abbreviate_names': False,
+                               'name_style': 'lastfirst' }
+          :sorting_style: author_year_title
+          :abbreviate_names: False
+          :name_style: lastfirst
+    """
+    required_arguments = 0
+    optional_arguments = 100
+    final_argument_whitespace = False
+    has_content = True
+
+    generator = None
+
+    def set_generator(generator):
+        Bibliography.generator = generator
+
+    def boolean(argument):
+        if argument == 'True': return True
+        if argument == 'False': return False
+        raise ValueError('must be boolean')
+    option_spec = {
+        'template': directives.unchanged,
+        'options': directives.unchanged,
+        'class': directives.class_option,
+        'filter_tag': directives.unchanged,
+        'pybtex_style_args': directives.unchanged,
+        'sorting_style': directives.unchanged,
+        'abbreviate_names': boolean,
```

Comment:
It is my understand that you can use `directives.flag` and can leave the above `boolean` method altogether.

```suggestion
        'abbreviate_names': directives.flag,
```

Reply:
`directives.flag` has a slightly different meaning: You wouldn't pass `True` or `False`. Instead, providing the argument (without value) means `True` and omitting the argument means `False`.

With `boolean` you can override the default value of `abbreviate_names` (or leave it as is). `directives.flag` would need a corresponding `do_not_abbreviate_names` flag to provide the same behaviour.